In [25]:
import pandas as pd
import numpy as np
new_times = pd.read_table('times.txt', header=None)

#### Assumptions:
These are bus trip times. <br/>
The old mean is 36 minutes, with a standard deviation of 6 minutes.<br/>
We are testing to see if the new bus is faster than the old bus.

#### Null Hypothesis:
$H_0$ :  $ {Time_{\mathrm{new}}} - {Time_{\mathrm{old}}} >= 0 $<br/>
The null hypothesis states that the new time will be greater than or equal to the old time.

#### Alternative Hypothesis:
$H_1$ :  $ {Time_{\mathrm{new}}} - {Time_{\mathrm{old}}} < 0 $<br/>
The alternative hypothesis states that the new time will be less than the old time.

In [26]:
print('new mean: {:,.2f} minutes'.format(new_times.mean()[0]))
print('new standard deviation: {:,.2f}'.format(new_times.std()[0]))

new mean: 34.47 minutes
new standard deviation: 7.14


#### Z-Test:
$Z = \frac{\mu_{old} - \mu_{new}}{\sigma_{old} / \sqrt{N}}$

In [30]:
mu_new = new_times.mean()[0]
mu_old = 36
std_old = 6
N = 100
Z = (mu_old - mu_new) / (std_old / np.sqrt(N))
print("Z-score: {:,.2f}".format(Z))

Z-score: 2.56


#### Conclusion:
The Z-score of 2.56 is above 2 standard deviations from the mean (at which p = 0.05). Because of this, we know that our p-value is < 0.05, and thus can reject the null hypothesis, and conclude that the new bus-route is indeed faster than the old route. 